android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java

```diff
@@ -122,18 +127,16 @@ protected void onResume() {
     }
 
     private void loadData() {
-        realmRepo.getSessionsByLocation(location)
-                .addChangeListener((sessions, orderedCollectionChangeSet) -> {
-                    this.sessions.clear();
-                    this.sessions.addAll(sessions);
-
-                    setUpcomingSession();
-                    sessionsListAdapter.setCopyOfSessions(sessions);
-                    sessionsListAdapter.notifyDataSetChanged();
-                    if (!Utils.isEmpty(searchText))
-                        sessionsListAdapter.filter(searchText);
-                    handleVisibility();
-                });
+        searchText = locationActivityViewModel.getSearchText();
```

Comment:
I want you to double check if the search is actually working correctly.  i would recommend you move this line below the line 101.
We don't need to get the string everytime from the ViewModel.

Reply:
Ok sure.